src/main.rs

```diff
@@ -651,6 +691,219 @@ fn should_initialize_git_repo(mut path: &Path) -> bool {
+}
+
+/// This functiomn will find the `toolchain` category in `template.yaml` and replace its placeholder option with one
+/// option per installed rustup toolchain that supports required target.
+fn populate_toolchain_category(
+    chip: Chip,
+    options: &mut [GeneratorOptionItem],
+    cli_toolchain: Option<&str>,
+    msrv: &check::Version,
+) -> Result<()> {
+    let target = chip.target().to_string();
+
+    let mut available = filter_toolchains_for(&target, msrv)?;
+
+    // for now, we should hide the generic toolchains for Xtensa (stable-*, beta-*, nightly-*).
+    if chip.is_xtensa() {
+        available.retain(|name| {
+            !(name.starts_with("stable") || name.starts_with("beta") || name.starts_with("nightly"))
+        });
+    }
+
+    // sanity check
+    if available.is_empty() {
+        if let Some(cli) = cli_toolchain {
+            if chip.is_xtensa()
+                && (cli.starts_with("stable")
+                    || cli.starts_with("beta")
+                    || cli.starts_with("nightly"))
+            {
+                bail!(
+                    "Toolchain `{cli}` is not supported for Xtensa targets; \
+                     please use different toolchain (e.g. `esp`)"
+                );
+            }
+
+            bail!(
+                "Toolchain `{cli}` does not have target `{target}` installed (or no toolchain does)"
+            );
+        }
+        log::warn!(
+            "No rustc toolchains found that have `{target}` installed; toolchain category will stay as placeholder"
+        );
+        return Ok(());
+    }
+
+    if let Some(cli) = cli_toolchain {
+        if !available.iter().any(|t| t == cli) {
+            if chip.is_xtensa()
+                && (cli.starts_with("stable")
+                    || cli.starts_with("beta")
+                    || cli.starts_with("nightly"))
+            {
+                bail!(
+                    "Toolchain `{cli}` is not supported for Xtensa targets; \
+                     please use an ESP toolchain (e.g. `esp`)"
+                );
+            }
+
+            bail!("Toolchain `{cli}` does not have target `{target}` installed");
+        }
+        // put CLI toolchain first in toolchain search case it was provided.
+        available.sort();
+        available.sort_by_key(|t| if t == cli { 0 } else { 1 });
+    }
+
+    // get default toolchain to mark it properly
+    let default: Option<String> = {
+        if let Ok(output) = Command::new("rustup")
+            .args(["show", "active-toolchain"])
+            .output()
+        {
+            if output.status.success() {
+                if let Some(line) = String::from_utf8_lossy(&output.stdout).lines().next() {
+                    line.split_whitespace().next().map(|name| name.to_string())
+                } else {
+                    None
+                }
+            } else {
+                None
+            }
+        } else {
+            None
+        }
+    };
```

Comment:
Maybe it will be handy to have a `get_default_toolchain`  function, it will also make `populate_toolchain_category` a bit simpler as it's quite big atm